src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -70,45 +74,57 @@ public void get(String key, final OnGetCompleteCallback callback) {
             final AsynchronousFileChannel afc =
                     AsynchronousFileChannel.open(path, StandardOpenOption.READ);
             channel = afc;
-            int headerSize = entry.getHeaderSize();
-            final int size = (int) file.length() - headerSize;
+            final int size = (int) file.length();
             final ByteBuffer buffer = ByteBuffer.allocate(size);
             afc.read(
                     /* destination= */ buffer,
-                    /* position= */ headerSize,
+                    /* position= */ 0,
                     /* attachment= */ null,
                     new CompletionHandler<Integer, Void>() {
                         @Override
                         public void completed(Integer result, Void v) {
                             closeChannel(afc, "completed read");
-                            // if the file size changes, return null
                             if (size != result) {
                                 VolleyLog.e(
                                         "File changed while reading: %s", file.getAbsolutePath());
-                                callback.onGetComplete(null);
+                                removeDeleteAndCall(key, callback, file);
```

Comment:
nit: I think this method name is unclear (remove what? delete what? call what?).

I think it's arguably implied that removing the file should imply it gets deleted from the map, so calling this something like "deleteFileAndInvokeCallback" would be a bit more readable.

Reply:
I changed it to that; it's definitely more descriptive.